jobs/build-charms/charms.py

```diff
@@ -974,9 +977,15 @@ def build(
             entity.push()
             entity.attach_resources()
             entity.promote(to_channel=to_channel)
+        except Exception:
+            entity.echo(traceback.format_exc())
+            failed_entities.append(entity)
         finally:
             entity.echo("Stopping")
 
+    if any(failed_entities):
+        raise BuildException("Encountered Charm Build Failure")
```

Comment:
We're not doing anything with the list of `failed_entities`.  Perhaps add a `len()` to this output so we know how many charms failed?  Or swap the list for a simple bool flag that tells us if at least one thing failed?

Reply:
i'm on it

Reply:
```bash
[canal] Stopping
Encountered 1 Charm Build Failure:
	canal
```